---
title: "MyNewProject"
author: "Divyanshi Srivastava, dvs5680@psu.edu"
output: html_document
date: June 8, 2016
---

```{r Normal}
x = rnorm(1000)
x = rnorm(1000,mean = 2, sd = 3)
#Note that the command takes sd and not var, which is usually used to descibe a normal distb. So lets label this
x = rnorm(1000,mean = 2, sd = 3)
hist(x, main = "N(2,9)", nclass = 50)
```

Exercise 1: 

Generate the phenotypic scores as 20 random normals. Call this vector pheno.

Generate the gene expression scores as a matrix with 1000 rows (genes) and 20 columns (samples). Call this matrix genes.

Compute the 1000 correlations between the phenotypic scores and the gene expression scores. Note that you will need to transpose the gene expression matrix to do this. Call the correlations cors.

```{r}
#creating a pheno score vector
pheno=rnorm(20)
#matrix of 1000 rows and 20 columns) So like 1000 gene exp values for the 20 samples.
genes=matrix(rnorm(1000*20), nrow = 1000, ncol = 20 )
#Computing cor
cors = cor(pheno, t(genes))
#plotting cors
hist(cors)
```


Note: Its nice to have the number of samples and number of genes, as variables on top so that if they change, you dont mess up the code. 


So.

```{r}
nsamples = 20
ngenes = 1000
```

Get the 10 most correlated genes:

```{r}
top = head(sort(abs(as.vector(cors)), decreasing = TRUE))
```

#The way correlation is computed, we do not need to force the simulated data to have a certain mean, as the mean is subtracted from all arguments. The SD, again does not need to be set.

#For other analyses. Such as clustering, or ANOVA. Then you'd need an estimate of the means/variances. 

#RNA-Seq data are not even close to being normal, so corelation doesnt make much sense.

#We need to force the rnorm generator to be the same each time if you're redoing the simuation. So like use the same seed. 



```{}
seesionInfo()
```